impeller/renderer/backend/vulkan/command_encoder_vk.cc

```diff
@@ -86,7 +80,7 @@ class TrackedObjectsVK {
 
  private:
   DescriptorPoolVK desc_pool_;
-  std::weak_ptr<CommandPoolVK> pool_;
+  std::shared_ptr<CommandPoolVK> pool_;
```

Comment:
Our command buffers always have a link to the command pool so it is never safe to have a dangling reference to a command buffer who is referring to a potentially dead pool.  This makes that guarentee.

Reply:
I would add this in a comment in the code (versus just here which will get lost to the sands of time)

Reply:
done